crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -1840,6 +1858,15 @@ async fn resolve_relative_request(
 
     let mut new_path = path_pattern.clone();
     if !options_value.fully_specified {
+        let fragment = fragment.await?;
+        if !fragment.is_empty() {
+            new_path.push(Pattern::Alternatives(
+                once(Pattern::Constant("".to_string()))
+                    .chain(once(Pattern::Constant(format!("#{fragment}"))))
```

Comment:
- this should also applied for fully specified requests
- in the case when we add the fragment to the path, we must no longer add it as fragment to the result